### PR TITLE
Add AWS Lambda fetcher to Asset Inventory

### PR DESCRIPTION
**Pull request by kubasobon**
### Summary of your changes

Add AWS Lambda fetcher to Asset Inventory. It works for the following resources:
- Functions and Function Aliases
- Layers
- Event Source Mappings

### Screenshot/Data

TBD

### Related Issues

Closes https://github.com/elastic/security-team/issues/9089

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- ~~I have added the necessary README/documentation (if appropriate)~~

#### Introducing a new rule?

No

### CODE REVIEW

File: internal/inventory/awsfetcher/fetcher_lambda.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		classification inventory.AssetClassification
+	}{
+		{"Lambda Event Source Mappings", s.provider.ListEventSourceMappings, inventory.AssetClassification{
+			Category:    inventory.CategoryInfrastructure,
+			SubCategory: inventory.SubCategoryIntegration,
+			Type:        inventory.TypeEventSource,
+			SubType:     inventory.SubTypeLambdaEventSourceMapping,
+		}},
+		{"Lambda Functions", s.provider.ListFunctions, newLambdaClassification(inventory.TypeServerless, inventory.SubTypeLambdaFunction)},
+		{"Lambda Layers", s.provider.ListLayers, newLambdaClassification(inventory.TypeServerless, inventory.SubTypeLambdaLayer)},
```
Comment (romulets) [inline internal/inventory/awsfetcher/fetcher_lambda.go:67]: Thinking, is there a value to have lambda layers not tied up to lambda functions?

File: internal/resources/providers/awslib/lambda/event_source_mapping.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+)
+
+type EventSourceMappingInfo struct {
+	EventSourceMapping types.EventSourceMappingConfiguration `json:"event_source_mapping_configuration"`
+	awsAccount         string
+	region             string
+}
+
+func (v EventSourceMappingInfo) GetResourceArn() string {
+	id := pointers.Deref(v.EventSourceMapping.UUID)
```
Comment (romulets) [inline internal/resources/providers/awslib/lambda/event_source_mapping.go:36]: Interesting the `UUID` contains an arn

Comment (romulets) [reply internal/resources/providers/awslib/lambda/event_source_mapping.go:36]: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/lambda@v1.54.6/types#EventSourceMappingConfiguration couldn't find any definition say it's or is not an ARN. Could you just confirm it? If it contains indeed ARN, I blame aws on bad naming :D

Comment (kubasobon) [reply internal/resources/providers/awslib/lambda/event_source_mapping.go:36]: It might not have a true ARN at all. I'll check.

Comment (kubasobon) [reply internal/resources/providers/awslib/lambda/event_source_mapping.go:36]: Ok, so it doesn't have an ARN per se.
```
aws --profile=cloud-security --region=us-east-1 lambda create-event-source-mapping --event-source arn:aws:sqs:us-east-1:378890115541:kuba-test.fifo --function-name kuba-test-func
{
    "UUID": "e4ee5f1b-53a3-4861-a892-d4153ac1e23b",
    "BatchSize": 10,
    "MaximumBatchingWindowInSeconds": 0,
    "EventSourceArn": "arn:aws:sqs:us-east-1:<ACCOUNT_ID>:kuba-test.fifo",
    "FunctionArn": "arn:aws:lambda:us-east-1:<ACCOUNT_ID>:function:kuba-test-func",
    "LastModified": "2024-06-17T13:21:47.009000+02:00",
    "State": "Creating",
    "StateTransitionReason": "USER_INITIATED",
    "FunctionResponseTypes": []
}
```

Just UUID. I think I'll make it return an empty ARN field, WDYT @romulets?

File: internal/inventory/awsfetcher/fetcher_lambda.go (hunk 1 above)
Comment (kubasobon) [reply internal/inventory/awsfetcher/fetcher_lambda.go:67]: image

File: internal/resources/providers/awslib/lambda/event_source_mapping.go (hunk 2 above)
Comment (romulets) [reply internal/resources/providers/awslib/lambda/event_source_mapping.go:36]: If we don't have an ARN available, I'd rather not return it. Avoids unintended usage and bugs

**Comment by kubasobon** [comment]
@romulets applied your remarks, PTAL :)

**Review by romulets — APPROVED** [review]

**Pull request merged**